src/main/java/org/folio/oaipmh/service/impl/InstancesServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package org.folio.oaipmh.service.impl;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.oaipmh.dao.InstancesDao;
+import org.folio.oaipmh.service.InstancesService;
+import org.folio.rest.jooq.tables.pojos.Instances;
+import org.folio.rest.jooq.tables.pojos.RequestMetadataLb;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+
+@Service
+public class InstancesServiceImpl implements InstancesService {
+
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+  private InstancesDao instancesDao;
+
+  public InstancesServiceImpl(InstancesDao instancesDao) {
+    this.instancesDao = instancesDao;
+  }
+
+  @Override
+  public Future<List<String>> cleanExpiredInstances(String tenantId, int expirationTimeSeconds) {
+    Promise<List<String>> promise = Promise.promise();
+    instancesDao.getExpiredRequestIds(tenantId, expirationTimeSeconds)
+      .onSuccess(ids -> {
+        List<Future> futures = new ArrayList<>();
```

Comment:
Sonar complains about code here as a code smell. Please check out.

Reply:
I've already discussed this with back-end guys, this is not needed o be fixed.